spring-integration-http/src/main/java/org/springframework/integration/http/outbound/ReactiveHttpRequestExecutingMessageHandler.java

```diff
@@ -0,0 +1,183 @@
+		this(new LiteralExpression(uri), webClient);
+		/*
+		 *  We'd prefer to do this assertion first, but the compiler doesn't allow it. However,
+		 *  it's safe because the literal expression simply wraps the String variable, even
+		 *  when null.
+		 */
+		Assert.hasText(uri, "URI is required");
+	}
+
+	/**
+	 * Create a handler that will send requests to the provided URI using a provided WebClient.
+	 * @param uriExpression A SpEL Expression that can be resolved against the message object and
+	 * {@link BeanFactory}.
+	 * @param webClient The WebClient to use.
+	 */
+	public ReactiveHttpRequestExecutingMessageHandler(Expression uriExpression, WebClient webClient) {
+		super(uriExpression);
+		this.webClient = (webClient == null ? WebClient.create() : webClient);
+		this.setAsync(true);
+	}
+
+	@Override
+	public String getComponentType() {
+		return (this.getExpectReply() ? "http:outbound-reactive-gateway" : "http:outbound-reactive-channel-adapter");
+	}
+
+	@Override
+	protected Object exchange(URI uri, HttpMethod httpMethod, HttpEntity<?> httpRequest, Object expectedResponseType) {
+		WebClient.UriSpec uriSpec;
+
+		// TODO use WebClient.method(HttpMethod) in the future version
+
+		switch (httpMethod) {
+			case GET:
+				uriSpec = this.webClient.get();
+				break;
+			case HEAD:
+				uriSpec = this.webClient.head();
+				break;
+			case POST:
+				uriSpec = this.webClient.post();
+				break;
+			case PUT:
+				uriSpec = this.webClient.put();
+				break;
+			case PATCH:
+				uriSpec = this.webClient.patch();
+				break;
+			case DELETE:
+				uriSpec = this.webClient.delete();
+				break;
+			case OPTIONS:
+				uriSpec = this.webClient.options();
+				break;
+			case TRACE:
+				throw new UnsupportedOperationException("WebClient doesn't support TRACE HTTP method");
+			default:
+				throw new UnsupportedOperationException("Unsupported HTTP method");
+		}
+
+		WebClient.HeaderSpec spec = uriSpec.uri(uri)
+				.headers(httpRequest.getHeaders());
+
+		Mono<ClientResponse> responseMono;
+		if (httpRequest.hasBody()) {
+			responseMono = spec.exchange(BodyInserters.fromObject(httpRequest.getBody()));
+		}
+		else {
+			responseMono = spec.exchange();
+		}
+
+		if (getExpectReply()) {
+
+			ResolvableType responseType;
+
+			if (expectedResponseType instanceof ParameterizedTypeReference<?>) {
+				responseType = ResolvableType.forType(((ParameterizedTypeReference<?>) expectedResponseType).getType());
+			}
+			else if (expectedResponseType != null) {
+				responseType = ResolvableType.forClass((Class<?>) expectedResponseType);
+			}
+			else {
+				responseType = null;
+			}
+
+			return responseMono.map(response ->
+					new ResponseEntity<>(responseType != null
+							? response.body(BodyExtractors.toMono(responseType)).block()
+							: null,
+							response.headers().asHttpHeaders(),
+							response.statusCode()))
+					.map(this::getReply);
+		}
+		else {
+			responseMono.block();
```

Comment:
H-m-m. Good catch, agree. Well, without blocking we don't have ability to handle error if that.
Although, I think we can bring here the same solution that we have for the `ListenableFuture` and its `onFailure()`. I mean that `sendErrorMessage()`.

Will fix soon!
Thank you!